src/components/ui/Dialog.tsx

```diff
@@ -0,0 +1,103 @@
+/**
+ * This file is based on the Dialog component from shadcn and customized for our needs.
+ *
+ * See the official docs for more info:
+ * shadcn/ui: https://ui.shadcn.com/docs/components/dialog
+ * Radix UI: https://www.radix-ui.com/themes/docs/components/dialog
+ */
+
+'use client';
+
+import { Icon } from '@/components/ui';
+import { cn } from '@/lib/tailwind/utils';
+
+import * as DialogPrimitive from '@radix-ui/react-dialog';
+import * as React from 'react';
+
+export const Root = DialogPrimitive.Root;
+export const Trigger = DialogPrimitive.Trigger;
```

Comment:
`Root` and `Trigger` are not used in this file.
These can be imported at each implement file directly from `react-dialog`. This might arise unnecessary dependency.

Reply:
After a discussion, it was concluded they should be imported/exported in this file to follow clean architecture principles.